src/createServer.js

```diff
@@ -1,11 +1,186 @@
 'use strict';
 
-// const express = require('express');
+const express = require('express');
+const cors = require('cors');
+const {
+  getAllUsers,
+  getUser,
+  createUserId,
+  addUser,
+  removeUser,
+  updateUser,
+} = require('./services/users.service');
+
+const {
+  getAllExpences,
+  getAllFilteredExpences,
+  getExpenceById,
+  validatePostData,
+  createExpenceId,
+  addExpence,
+  removeExpence,
+  updateExtence,
+} = require('./services/expences.service');
+const app = express();
 
 function createServer() {
-  // Use express to create a server
-  // Add a routes to the server
-  // Return the server (express app)
+  app.use(cors());
+
+  app.get('/users', (req, res) => {
+    res.send(getAllUsers());
+  });
+
+  app.get('/users/:id', (req, res) => {
+    const { id } = req.params;
+
+    if (!id) {
+      res.sendStatus(400);
+
+      return;
+    }
+
+    if (!getUser()) {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    res.send(getUser());
+  });
+
+  app.post('/users', express.json(), (req, res) => {
+    const { name } = req.body;
+
+    if (!name) {
+      res.sendStatus(400);
+
+      return;
+    }
+
+    const user = {
+      id: createUserId(),
+      name,
+    };
+
+    addUser(user);
+
+    res.statusCode = 201;
+    res.send(user);
+  });
+
+  app.delete('/users/:id', (req, res) => {
+    const { id } = req.params;
+
+    const usersLength = getAllUsers().length;
+
+    removeUser(id);
+
+    if (getAllUsers().length === usersLength) {
```

Comment:
First, you need to find out if such a user exists, and only then delete it, not the other way around